src/transformStateWithClones.js

```diff
@@ -8,6 +8,48 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+  const ADD_PROPERTIES = 'addProperties';
+  const REMOVE_PROPERTIES = 'removeProperties';
+  const CLEAR_LIST = 'clear';
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ADD_PROPERTIES:
+        addProperties(stateCopy, action.extraData);
+        stateHistory.push({ ...stateCopy });
```

Comment:
you doing it in every case, better to move it after switch case for prevent repeat code